add lab5 file and info

1ovann opened the pull request:
1. added some information about lab5 name and task
2. added file with a finished code of lab5

1ovann committed:
add info about lab5

1ovann committed:
add lab5 file with finished code

shirouex commented on lab5.py:
```diff
@@ -0,0 +1,61 @@
+class File:
```
add docstring

1ovann replied:
added

shirouex commented on lab5.py:
```diff
@@ -0,0 +1,61 @@
+class File:
+    # pylint: disable=R0903
+    def __init__(self, name, extension, size):
```
add docstring

1ovann replied:
added

shirouex commented on lab5.py:
```diff
@@ -0,0 +1,61 @@
+        self.extension = extension
+        self.size = size
+
+    def __str__(self):
```
add docstring

1ovann replied:
added

shirouex commented on lab5.py:
```diff
@@ -0,0 +1,61 @@
+        return f"{self.name}.{self.extension} ({self.size} bytes)"
+
+
+class Folder:
```
add docstring

1ovann replied:
added

shirouex commented on lab5.py:
```diff
@@ -0,0 +1,61 @@
+
+
+class Folder:
+    def __init__(self, name):
```
add docstring

1ovann replied:
added

shirouex commented on lab5.py:
```diff
@@ -0,0 +1,61 @@
+        self.files = []
+        self.folders = []
+
+    def add_file(self, file):
```
add docstring

1ovann replied:
added

shirouex commented on lab5.py:
```diff
@@ -0,0 +1,61 @@
+    def add_file(self, file):
+        self.files.append(file)
+
+    def add_folder(self, folder):
```
add docstring

1ovann replied:
added

shirouex commented on lab5.py:
```diff
@@ -0,0 +1,61 @@
+        for file in self.files:
+            print("  " * (indent + 1) + str(file))
+        for folder in self.folders:
+            folder.print_contents(indent + 1)
```
1. add docstring
2. Too complicated code, increase readability

1ovann replied:
1. added.
2. tried to simplify it by removing multiplication and replacing parameter. looks much easier for me rn.

shirouex replied:
Incredible, now it seems much easier for me.

shirouex commented on lab5.py:
```diff
@@ -0,0 +1,61 @@
+        for folder in self.folders:
+            folder.print_contents(indent + 1)
+
+    def get_longest_path(self):
```
add docstring

1ovann replied:
added

shirouex commented on lab5.py:
```diff
@@ -0,0 +1,61 @@
+
+
+def main():
+    file1 = File("file1", "txt", 1024)
+    file2 = File("file2", "doc", 2048)
+    file3 = File("file3", "jpg", 3072)
```
use more descriptive names, it will enhance your code readability. You should name like text_file, document_file, image_file.

1ovann replied:
its an instance of class "file" and its name can be defined as you wish. i dont think renaming them is gonna bring any big difference, because we can identify files by their extension.

shirouex replied:
Yes, there is no purpose in doing so. You can simply identify this by their extension.

shirouex commented on lab5.py:
```diff
@@ -0,0 +1,61 @@
+        self.size = size
+
+    def __str__(self):
+        return f"{self.name}.{self.extension} ({self.size} bytes)"
```
Method __str__ should provide more details.

1ovann replied:
what else should it represent? i think file name, extension and size are enough

shirouex replied:
Yes, method str represents all necessary information, so you don't need to add any changes.

shirouex reviewed (changes requested):
Improve few methods and add docstrings

1ovann committed:
add docstrings, longest_path fix, tree improve

1. added docstring to each class and method.
2. completely changed the get_longest_path method. the previous one had a lot of bugs and worked incorrectly in some conditions, now everything works fine.
3. replaced argument "indent" by "prefix" and removed multiplication. now it looks much easier

shirouex reviewed (approved):
All necessary changes were added, perfect code.
